contrib/ci/Jenkinsfile.mpi

```diff
@@ -105,6 +105,7 @@ pipeline
                     -D CMAKE_BUILD_TYPE=Debug \
                     -D DEAL_II_WITH_MPI=ON \
                     -D DEAL_II_UNITY_BUILD=OFF \
+                    -D DEAL_II_COMPILER_HAS_FUSE_LD_GOLD=false \
```

Comment:
That means we would be running into this underlining issue also in user code which could be seen as a regression...

Reply:
It was mere coincidence that our "switch through linker logic" actually duct taped over this problem.

The problem here is that the CI image installs a Trilinos library that is not linked correctly. It is mere coincidence that ld.bfd is forgiving enough to actually link the library. For example, ld.lld is much, much stricter in what it considers valid or not.

The basic problem with any change to our CMake infrastructure is the fact that its intended and observed API is used in so many ways that it would not be possible to change anything without risking a "regression" somewhere.
The issue why I haven't put the "switch through linker logic" back is because it is more complex to set it up with interface and imported targets and I needed some flexibility to port the build system.

I suggest we try to get this logic back in a follow-up pull request that also addresses the missing test for the full link interface.